homeassistant/components/shelly/config_flow.py

```diff
@@ -38,18 +39,17 @@ async def validate_input(hass: core.HomeAssistant, host, data):
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
     options = aioshelly.ConnectionOptions(
-        host, data.get(CONF_USERNAME), data.get(CONF_PASSWORD)
+        gethostbyname(host), data.get(CONF_USERNAME), data.get(CONF_PASSWORD)
     )
-    coap_context = await aiocoap.Context.create_client_context()
+    await get_coap_context(hass)
```

Comment:
this makes no sense. `get_coap_context` should return the context. 
```suggestion
    context = await get_coap_context(hass)
```